### ISSUE TITLE
Doubt about Training attribute prediction

### ISSUE THREAD

**Issue by hackergod010** [open-mmlab/mmfashion#49]
Can we train an attribute prediction model without the landmarks file? 

This doubt also arises cause of the fact that [here](https://github.com/open-mmlab/mmfashion/blob/49649a2ac3b5362fe4100095edb989d9b73889cb/mmfashion/datasets/Attr_Pred.py#L67) `self.landmarks = None` but [here](https://github.com/open-mmlab/mmfashion/blob/49649a2ac3b5362fe4100095edb989d9b73889cb/mmfashion/datasets/Attr_Pred.py#L94), in the `get_basic_item` function, `origin_landmark = self.landmarks[idx]` is used.

If at all we do really need the landmarks for training how do we save the predictions from the landmark detector into the format required for training attribute prediction model?

**Comment by veralauee** [comment]
Yes, you can train the attribute prediction model without landmark file, use "config/attribute_predict/global_predictor*" 
I modified the code and make the landmark file as None, also modified the code you mentioned.
Thanks for your advise.

**Comment by hackergod010** [comment]
What are the landmarks used for though? Does the model give better predictions with landmarks? 

Also what is the format followed by the output of the `test_landmark.py` file?

**Comment by veralauee** [comment]
The landmarks are used in "RoI_Pooling" method. Specifically, it will extract the local feature around the landmark of the cloth, e.g., knees, shoulders. Thus, it can improve the prediction compared to "Global_Pooling".
"test_landmark.py" is used for landmark prediction. It will output the coordinate(x and y) for the key point in the cloth.

**Comment by hackergod010** [comment]
I was able to prepare my dataset in a similar way, and was able to run the `prepare_attr_pred.py`.

The dataset has 9 categories and 213 attributes, however I've run into some issues while training the model.
When I run the train script for attribute prediction I get the following: 

```
$ python mmfashion/tools/train_predictor.py --config config.py
2020-05-08 22:27:45,033 - INFO - Distributed training: False
model built
unexpected key in source state_dict: meta, state_dict, optimizer

missing keys in source state_dict: layer3.3.bn2.running_mean, layer2.3.bn1.running_var, layer3.4.bn3.running_var, layer3.0.bn1.running_var, layer2.0.bn1.running_mean, layer4.1.bn2.weight, layer3.1.bn1.num_batches_tracked, layer3.1.conv3.weight, layer4.0.bn1.running_var, layer3.4.bn1.weight, layer4.0.bn2.running_var, layer3.3.bn3.bias, layer1.1.bn3.running_mean, layer2.2.bn2.bias, layer3.5.bn3.running_var, layer3.2.bn1.weight, layer2.1.bn2.weight, layer1.1.bn2.running_var,
layer2.0.downsample.1.num_batches_tracked, layer2.2.conv2.weight, layer2.1.bn2.num_batches_tracked, layer4.0.downsample.1.running_var, bn1.bias, layer2.3.conv1.weight, layer3.4.bn2.num_batches_tracked, layer2.1.conv3.weight, layer4.0.downsample.0.weight, layer3.0.bn3.num_batches_tracked, layer3.5.conv3.weight, layer1.0.conv1.weight, layer4.2.bn1.running_mean, layer2.1.bn3.bias, layer1.2.bn2.num_batches_tracked, layer3.1.bn1.bias, layer3.0.downsample.1.weight, layer1.2.bn2.weight,
layer3.1.bn3.num_batches_tracked, layer3.3.bn3.running_var, layer1.1.bn1.num_batches_tracked, layer1.0.bn3.running_mean, layer3.3.conv1.weight, layer2.2.bn3.num_batches_tracked, layer2.3.bn1.running_mean, layer3.3.bn2.num_batches_tracked, layer4.0.conv2.weight, layer3.0.bn3.running_mean, layer1.2.conv1.weight, layer3.3.bn2.bias, layer2.3.bn2.running_var, layer1.2.bn1.running_mean, layer3.0.bn3.running_var, layer3.4.bn1.running_mean, layer3.0.conv1.weight, layer4.0.downsample.1.bias,  
layer4.2.conv3.weight, layer3.5.conv1.weight, layer4.1.bn3.num_batches_tracked, layer3.1.bn1.running_mean, layer1.2.bn1.weight, layer1.1.bn1.running_var, layer3.3.bn1.weight, layer2.0.bn1.running_var, layer3.2.bn2.weight, layer2.3.conv3.weight, layer2.0.bn1.bias, layer3.0.bn2.running_mean, layer2.1.conv1.weight, layer2.1.bn2.running_var, layer4.0.downsample.1.num_batches_tracked, layer2.3.conv2.weight, layer1.1.bn2.num_batches_tracked, layer2.1.bn3.weight, layer2.1.bn3.running_var,
layer4.1.conv1.weight, layer2.3.bn2.weight, layer4.0.bn2.weight, layer4.1.bn3.running_var, layer3.4.bn2.bias, layer3.5.bn1.num_batches_tracked, layer4.1.bn1.bias, layer3.3.bn2.weight, layer3.2.bn1.num_batches_tracked, layer4.2.bn3.num_batches_tracked, layer4.2.bn3.weight, layer1.1.bn1.bias, layer3.0.bn3.bias, layer1.0.bn2.running_mean, layer2.0.downsample.1.running_var, layer4.2.conv2.weight, layer2.1.bn1.running_mean, layer3.0.bn3.weight, layer3.1.bn3.running_mean, layer2.1.conv2.weight,
layer1.2.bn3.running_var, layer2.0.bn2.running_mean, layer3.5.bn2.weight, layer4.0.bn1.bias, layer3.4.bn3.running_mean, layer1.0.bn1.running_mean, layer1.2.bn1.bias, layer2.2.conv3.weight, layer3.1.bn3.bias, layer2.0.conv3.weight, layer3.1.conv1.weight, layer3.2.bn3.running_mean, layer2.2.bn3.bias, layer2.2.bn2.running_mean, layer4.0.downsample.1.running_mean, layer3.5.bn3.num_batches_tracked, layer3.5.bn2.num_batches_tracked, layer2.0.bn2.num_batches_tracked, layer2.2.bn1.running_mean,
layer4.2.bn2.weight, layer2.1.bn3.num_batches_tracked, layer2.3.bn1.weight, layer1.0.downsample.1.bias, layer2.3.bn2.num_batches_tracked, layer4.2.bn1.bias, layer3.1.bn2.bias, layer1.0.bn1.weight, layer4.2.bn3.running_var, layer3.1.bn3.weight, layer3.2.bn2.running_var, layer4.0.bn3.running_var, layer1.1.conv2.weight, layer1.2.conv3.weight, layer4.0.bn2.running_mean, layer3.1.bn2.running_mean, layer2.3.bn3.weight, layer3.0.bn2.bias, layer4.1.bn1.running_mean, layer1.1.conv1.weight,
layer3.2.conv1.weight, layer1.1.bn2.running_mean, bn1.running_mean, layer3.2.bn1.bias, layer1.0.bn2.num_batches_tracked, layer3.4.conv3.weight, layer1.0.downsample.1.running_var, layer2.0.bn3.running_mean, layer3.3.bn1.num_batches_tracked, layer1.0.downsample.1.running_mean, layer2.1.bn1.weight, layer1.2.bn2.bias, layer3.2.bn3.running_var, layer4.0.bn1.num_batches_tracked, layer3.5.bn2.running_var, layer2.2.bn1.bias, layer1.2.bn3.running_mean, bn1.running_var, layer2.2.bn1.weight,
layer3.4.bn2.weight, layer4.1.bn1.weight, layer3.3.bn1.running_var, layer1.2.conv2.weight, layer2.3.bn3.running_var, layer3.2.bn3.weight, layer1.1.bn3.num_batches_tracked, layer2.0.bn3.bias, layer3.5.bn3.running_mean, layer3.5.bn1.running_mean, layer4.1.conv3.weight, layer4.0.downsample.1.weight, layer3.2.conv3.weight, layer3.4.bn3.num_batches_tracked, layer1.2.bn2.running_mean, layer3.3.bn3.running_mean, layer1.2.bn3.weight, layer2.2.bn2.weight, layer1.0.bn2.weight,
layer3.4.bn1.running_var, layer4.1.bn1.running_var, layer1.1.bn3.weight, layer2.3.bn1.bias, layer3.2.bn1.running_var, layer3.0.bn2.num_batches_tracked, layer2.0.bn3.weight, layer4.0.bn1.weight, layer2.0.conv1.weight, layer4.1.bn2.running_mean, layer3.4.conv2.weight, layer2.0.bn2.bias, layer2.0.bn2.running_var, layer3.3.bn1.bias, layer3.1.bn1.weight, layer1.1.bn2.bias, layer3.0.bn1.num_batches_tracked, layer2.0.downsample.1.running_mean, layer3.2.bn2.running_mean, layer3.4.conv1.weight,
layer3.1.conv2.weight, layer2.2.bn3.weight, layer3.0.downsample.1.running_mean, layer3.0.downsample.1.bias, layer3.3.conv2.weight, layer3.3.bn3.weight, layer1.2.bn2.running_var, layer3.2.bn2.bias, layer3.3.bn2.running_var, layer1.1.bn1.running_mean, layer1.1.bn3.bias, layer1.0.bn3.bias, layer2.0.downsample.0.weight, layer3.2.bn1.running_mean, layer3.5.bn1.running_var, layer3.3.bn1.running_mean, layer2.0.bn2.weight, layer2.0.bn3.num_batches_tracked, layer3.1.bn1.running_var,
bn1.num_batches_tracked, layer1.0.downsample.1.weight, layer1.0.bn3.num_batches_tracked, layer3.0.bn2.weight, layer4.0.bn1.running_mean, layer4.2.bn1.running_var, layer3.5.bn3.bias, layer3.0.bn1.running_mean, layer4.2.bn1.weight, layer3.2.bn3.bias, layer1.2.bn1.running_var, layer4.1.bn3.bias, layer2.3.bn3.bias, layer4.0.bn2.num_batches_tracked, layer4.1.bn2.bias, layer4.0.bn2.bias, layer2.0.downsample.1.bias, layer2.2.bn2.running_var, layer4.2.bn1.num_batches_tracked,
layer3.0.downsample.0.weight, layer2.1.bn2.bias, layer3.1.bn3.running_var, layer1.2.bn1.num_batches_tracked, layer2.1.bn3.running_mean, layer1.0.conv3.weight, layer3.1.bn2.num_batches_tracked, layer3.4.bn1.num_batches_tracked, bn1.weight, layer3.5.bn2.bias, layer4.0.bn3.weight, layer2.3.bn1.num_batches_tracked, layer1.0.downsample.0.weight, layer1.0.bn3.weight, layer4.1.bn1.num_batches_tracked, layer4.2.bn2.running_var, layer4.2.bn3.running_mean, conv1.weight,
layer4.0.bn3.num_batches_tracked, layer4.2.bn2.num_batches_tracked, layer1.2.bn3.bias, layer1.0.bn3.running_var, layer2.1.bn2.running_mean, layer2.1.bn1.num_batches_tracked, layer2.0.bn3.running_var, layer4.1.bn3.running_mean, layer3.0.downsample.1.num_batches_tracked, layer4.1.bn2.num_batches_tracked, layer3.4.bn3.bias, layer1.1.bn3.running_var, layer2.3.bn2.bias, layer3.2.bn3.num_batches_tracked, layer1.1.bn1.weight, layer2.2.bn3.running_mean, layer1.0.downsample.1.num_batches_tracked,
layer1.0.bn2.bias, layer4.2.bn2.bias, layer1.0.bn1.num_batches_tracked, layer4.1.conv2.weight, layer3.3.conv3.weight, layer3.0.bn2.running_var, layer1.0.bn1.running_var, layer3.5.bn1.bias, layer3.3.bn3.num_batches_tracked, layer3.5.bn3.weight, layer2.1.bn1.bias, layer1.2.bn3.num_batches_tracked, layer2.2.conv1.weight, layer3.4.bn3.weight, layer3.2.bn2.num_batches_tracked, layer3.0.conv3.weight, layer2.0.conv2.weight, layer4.0.bn3.bias, layer3.2.conv2.weight,
layer2.2.bn3.running_var,layer1.0.bn1.bias, layer1.1.conv3.weight, layer4.2.bn3.bias, layer3.1.bn2.weight, layer3.1.bn2.running_var, layer2.0.bn1.num_batches_tracked, layer2.2.bn1.running_var, layer3.0.conv2.weight, layer4.0.bn3.running_mean, layer2.3.bn2.running_mean, layer3.5.bn1.weight, layer4.0.conv3.weight, layer1.1.bn2.weight, layer4.2.conv1.weight, layer4.2.bn2.running_mean, layer3.4.bn2.running_mean, layer1.0.bn2.running_var, layer2.2.bn2.num_batches_tracked, layer3.4.bn1.bias,
layer3.5.bn2.running_mean, layer2.0.bn1.weight, layer3.0.downsample.1.running_var, layer3.0.bn1.bias, layer3.4.bn2.running_var, layer4.1.bn2.running_var, layer3.5.conv2.weight, layer3.0.bn1.weight, layer2.3.bn3.num_batches_tracked, layer2.1.bn1.running_var, layer2.2.bn1.num_batches_tracked, layer1.0.conv2.weight, layer2.0.downsample.1.weight, layer4.0.conv1.weight, layer4.1.bn3.weight, layer2.3.bn3.running_mean

dataset loaded
dataloader built
model paralleled
2020-05-08 22:27:59,399 - INFO - Start running, host: d0, work_dir: checkpoint/Predict/resnet/global
2020-05-08 22:27:59,399 - INFO - workflow: [('train', 40)], max: 4 epochs
.conda/envs/open-mmlab/lib/python3.7/site-packages/torch/nn/parallel/_functions.py:61: UserWarning: Was asked to gather along dimension 0, but all input tensors were scalars; will instead unsqueeze and return a vector.
  warnings.warn('Was asked to gather along dimension 0, but all '
2020-05-08 22:28:05,558 - INFO - Epoch [1][10/2908]     lr: 0.00012, eta: 1:58:12, time: 0.610, data_time: 0.136, memory: 8636, loss_attr: 0.9555, loss: 0.9555
```

I was wondering what the error was here and how to load the pretrained model in the MODEL ZOO to train a custom model. 
The loss also does not decrease past `~0.6904` and it reaches this within the first epoch remains the same till 40. 
This remains true even if I don't initialize weights/use a pretrained model.

Also where does the model use categories since attribute prediction is only what interests us and the images have already been classified into their various categories as can be seen from their filepath?

In your docs you have mentioned that the images are resized to `(300, 300)` but all the configs specify an image size of `(224, 224)`, is this right?

**Comment by veralauee** [comment]
We just added "category prediction" function, please refer to [model_zoo](https://github.com/open-mmlab/mmfashion/blob/master/docs/MODEL_ZOO.md#category-and-attribute-predictionfine) and [user guide](https://github.com/open-mmlab/mmfashion/blob/master/docs/GETTING_STARTED.md).

Also, this updated category and attribute predictor is more accurate then the old one.

You need to initialize the backbone using the ImageNet pretrained weights, otherwise the performance will be hurt. 
All the images are resized to (224,224) in the default config. However, you can change it in the config file as you need.
